server_modules/scores.js

```diff
@@ -0,0 +1,49 @@
+var utils = require('./utils');
+var fileSystem = require('./file_system');
+
+function filterPublished(score) {
+    return score.published;
+}
+
+function reduceToMap(key) {
+    return function(arr) {
+        return arr.reduce(function(map,record) {
+            map[record[key]] = record;
+            return map;
+        },{});
+    }
+}
+
+exports.route = function(app) {
+
+    //get all, grouped by round
+    app.get('/scores/',function(req,res) {
+        Q.all([
+            fileSystem.parseDataFile('scores.json'),

```

Comment:
Actually, the main reason for the function, is that it returns a Promise for the parsed data, such that it is very easy to use in a Promise chain: `fs.readFile(...).then(parseJSON).then(doSomethingWithTheObject).catch(handleIOorJSorSomethingErrors)`